rfc/0003-tests-arch.md

```diff
@@ -0,0 +1,188 @@
+repository (e.g. push, pull request).
+
+
+## End-to-end tests
+
+These are the most complex tests. They will simulate what Kubewarden users will
+do to install, upgrade, configure and possibly uninstall Kubewarden stack.
+
+In these tests, different components versions would be tested together.
+Therefore, ensuring that the user will not face issues when using different
+components configuration and versions together. This should ensure that
+Kubernetes operator will have the expected features working.
+
+These tests are trigger when we have new releases (or release candidate) of
+any of the Kubewarden stack.  It should follow some steps:
+
+1. Setup Kubernetes cluster
+  Kubewarden is expected to run in multiple Kubernetes versions and machine
+  architectures. So, the end-to-end tests should ensure that all the supported
+  versions and architectures are tested. The initial versions of the tests can
+  be simple in a single arch and version.
+2. Install Kuberwarden stack on the cluster from step 1
+  The version to be tested should be installed. This means that when a component
+  is released. It should be tested against all the other components versions which
+  it supposed to work together.
+3. Run basic end-to-end tests
+  Validate if the basic features are working as expected
+4. Run reconfiguration tests
+  Validates if the user is able to reconfigure a Kubewarden stack installed
+5. Run upgrade tests
+  Validate if the user is able to upgrade and continue to work with the latest release.
+
+Steps 3,4 and 5 may run in parallel when necessary (e.g. time to execute all tests
+is to long)
+
+Considering the tests will validate multiple components of the Kubewarden stack
+together, it desirable to keep all the end-to-end test files in a separated
+repository. This repository should contains all the yaml files, scripts, and
+helper code to run the tests. This also helps to avoid duplicating tests that
+validate some feature that impacts multiple components.
+
+The end-to-end tests repository should be independent. Which means that it should
+run the tests independently of the tests and development cycle of the other
+Kubewarden repositories. But every time a new component release is available
+for testing, the end-to-end tests should be trigger automatically with no human
+intervention.  Besides that, running the end-to-end test on demand and in pre defined
+periods is also desirable.
+
+We could also configure the end-to-end tests to run against the latest version of
+all Kubewarden components. Thus, spotting some possible issues before a official
+release candidate.
+
+Considering the high workload nature of the end-to-end tests it may be necessary
+run them in a dedicated runner outside from Github hosted runners. From past
+experiences, the runners from Github could be slow and the tests failed due
+timeout issues.
+
+This tests can be written using bash and frameworks similar to [bats](https://github.com/bats-core/bats-core/).
```

Comment:
I understand the concern and I agree with reling on a specific tools. I've proposed bats because two main reasons:
1. We already use it in some repositories
2. Some time ago, we was talking about writing the tests using bash. Even if I'm not big fan of bash for this, if we decided to follow this path, we will need tools already created by bats. So, we will not reinvent the weel.

However, if we could use the Rancher testing infrastructure we can check how they run their tests and use the same tools. Thus, we can use the same infra already in place (considering that we have access to this infra).

Reply:
Ok, let's leave `bats` in the document. Please merge the grammar fixes :heart: